From 6a56fc1b9818c8724285a76e1086d37de7150079 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 27 Feb 2024 15:11:00 +0100
Subject: [PATCH 0963/1695] dts: stm32mp2: add CRYP2 node in device tree
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since the stm32mp25xf and stm32mp25xc can have CRYP2 available, it
should be added to the device tree.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 82a49ddf2f81e736b18bc6187ea51fa614bdf6ff dts: stm32mp2: add CRYP2 node in device tree]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: Icd6d6fb00c4261e8d43578fd09d28330775d14ac
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363263
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/386330
---
 core/arch/arm/dts/stm32mp25xc.dtsi | 9 +++++++++
 core/arch/arm/dts/stm32mp25xf.dtsi | 9 +++++++++
 2 files changed, 18 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp25xc.dtsi b/core/arch/arm/dts/stm32mp25xc.dtsi
index c752f7916..04c3757dc 100644
--- a/core/arch/arm/dts/stm32mp25xc.dtsi
+++ b/core/arch/arm/dts/stm32mp25xc.dtsi
@@ -15,6 +15,15 @@
 		status = "disabled";
 	};
 
+	cryp2: cryp@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x400>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		status = "disabled";
+	};
+
 	saes: saes@42050000 {
 		compatible = "st,stm32mp13-saes";
 		reg = <0x42050000 0x400>;
diff --git a/core/arch/arm/dts/stm32mp25xf.dtsi b/core/arch/arm/dts/stm32mp25xf.dtsi
index d879fd4c6..ba9aa41ed 100644
--- a/core/arch/arm/dts/stm32mp25xf.dtsi
+++ b/core/arch/arm/dts/stm32mp25xf.dtsi
@@ -15,6 +15,15 @@
 		status = "disabled";
 	};
 
+	cryp2: cryp@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x400>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		status = "disabled";
+	};
+
 	saes: saes@42050000 {
 		compatible = "st,stm32mp13-saes";
 		reg = <0x42050000 0x400>;
-- 
2.47.2

